addons/mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -24,13 +24,82 @@
             var self = this;
             this._super();
             this.choose_mailing_list('click').fail(function () {
-                self.editor.on_remove();
+                self.editor.on_remove($.Event( "click" ));
             });
         },
         clean_for_save: function () {
             this.$target.addClass("hidden");
         },
     });
+
+    website.snippet.options.newsletter_popup = website.snippet.Option.extend({
+        select_mailing_list: function (type, value) {
```

Comment:
What is the purpose of value (that is not used), and type ? Do you really have some logic associated to it ?